components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java

```diff
@@ -2003,7 +2006,13 @@ void deleteObjects(SecurityContext ctx, List<IObject> objects)
         try {
             DataManagerFacility dmf = gw.getFacility(DataManagerFacility.class);
             CmdCallbackI cb = dmf.delete(ctx, objects);
-            cb.loop(100, 250);
+            Response res = cb.loop(100, 250);
+            if (res instanceof GraphException) {
+                GraphException ge = (GraphException) res;
+                throw new Exception("Cannot delete the object: "+ ge.message);
+            }
+            if (res instanceof ERR)
+                throw new Exception("Cannot delete the object.");
```

Comment:
The exception will be wrapped into a DSAccessException in the catch clause anyway. I know, the whole exception handling is pretty messy, but I don't know an easy way to clean that up.